pkg/client/rootd/dns/client_queue.go

```diff
@@ -0,0 +1,47 @@
+package dns
```

Comment:
Since this only applies to the overriding resolver, and only on Linux (no other resolver will ever use a fallback), it would be better if all of this was conditioned using file names with a `_linux.go` suffix.

Reply:
I'm not sure it's possible because server.go relies on storing the connection pool

Reply:
So, does that mean that all DNS implementations will create 10 connections even though they are never used?

Reply:
oh no -- it's nil everywhere except the _linux implementations, i just meant that the type has to be there for the server to compile

Reply:
Ah, I see. But couldn't the type just be a small interface then? Seems a bit wasteful to drag the implementation along when it's never used.